api/src/main/java/uk/gov/dvsa/mot/trade/read/core/TradeReadServiceDatabase.java

```diff
@@ -264,6 +278,54 @@ private DisplayMotTestItem mapMotTestToDisplayMotTestItem(MotTest motTest, Vehic
         return displayMotTestItem;
     }
 
+    private uk.gov.dvsa.mot.trade.api.Vehicle getLatestDvlaVehicleAndMapToTradeVehicle(List<DvlaVehicle> vehicles) {
+
+        uk.gov.dvsa.mot.trade.api.Vehicle tradeVehicle = null;
+
+        if (vehicles.isEmpty()) {
+            return null;
+        }
+
+        if (!CollectionUtils.isNullOrEmpty(vehicles)) {
+            DvlaVehicle dvlaVehicle = vehicles.get(0);
+
+            for (DvlaVehicle dvlaVehicle1 : vehicles) {
+
+                if (dvlaVehicle1.getLastUpdatedOn().after(dvlaVehicle.getLastUpdatedOn())) {
```

Comment:
It does allow nulls, but this is entered by mysql through triggers so not something we would ever expect to be null. But null checks can surround everything to be safe.